cli/azd/pkg/project/project_config.go

```diff
@@ -291,3 +224,16 @@ func LoadProjectConfig(projectPath string) (*ProjectConfig, error) {
 	projectConfig.Path = filepath.Dir(projectPath)
 	return projectConfig, nil
 }
+
+// Gets the current project config from the IoC container
+// This method ensures the same instance is returned that can be referenced from any component
+// ex) Can be referenced from within middleware as well as command actions
+// Returns and error when the project is not found
+func GetCurrent() (*ProjectConfig, error) {
+	var projectConfig *ProjectConfig
+	if err := ioc.Global.Resolve(&projectConfig); err != nil {
```

Comment:
If I understood the intention correctly, I think this is to avoid the problem with `azdContext` not always available via injection. 

One straightforward but maybe "hacky" way is with pointer indirection, we can have `init` update the actual instance of `azdContext` and `projectConfig`:

1. In dependency registration, register a `nil` return instead of `azdcontext.NewContext` as the provider function for `up`
1. Update `init` to set the pointer value of `azdContext`

Reply:
This allows us to share the same `ProjectConfig` instance between middleware and the action via the container's singleton instance.  I will look into the pointer resetting but don't want to block on this PR.

Reply:
It's actually to allow a single instance to be shared in middleware & our actions via the container.  I will look into if/how we could do the pointer reassignment but hope not to block on this for the time being.